tket2/src/portmatching/indexing.rs

```diff
@@ -68,104 +49,63 @@ pub enum HugrVariableValue {
     OutgoingPort(hugr::Node, hugr::OutgoingPort),
 }
 
-impl HugrVariableValue {
-    fn node(&self) -> hugr::Node {
-        match *self {
-            HugrVariableValue::Node(node) => node,
-            HugrVariableValue::IncomingPort(node, _) => node,
-            HugrVariableValue::OutgoingPort(node, _) => node,
-        }
-    }
-}
-
 /// A map to store bindings for variables in a hugr.
 pub type HugrBindMap = BTreeMap<HugrVariableID, HugrVariableValue>;
 
-/// A port variable ID, given based on the unique IDs given to incoming ports.
-///
-/// - An incoming port ID is given by a path from a root node to an incoming port.
-/// - An outgoing port ID is given by the opposite incoming port ID. This defines
-///   the outgoing port uniquely as there is a one-to-many outgoing port to
-///   incoming port relationship.
+/// A port variable ID, given by a node ID and a port offset.
 #[derive(
-    Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, serde::Serialize, serde::Deserialize,
+    Clone, Copy, Debug, Eq, PartialOrd, Ord, Hash, PartialEq, serde::Serialize, serde::Deserialize,
 )]
-pub enum HugrPortID {
-    /// A port variable that binds to an outgoing port.
-    Outgoing {
-        /// The path from the root node to an incoming port opposite.
-        opposite_port: HugrPath,
-    },
-    /// A port variable that binds to an incoming port.
-    Incoming {
-        /// The path from the root node to the incoming port.
-        path_from_root: HugrPath,
-    },
+pub struct HugrPortID {
+    /// The node that the port belongs to.
+    pub node: HugrNodeID,
+    /// The port offset.
+    pub port: hugr::Port,
 }
 
 impl HugrPortID {
+    pub(super) fn new(node: HugrNodeID, port: hugr::Port) -> Self {
```

Comment:
Why is this not public?
The internals are `pub`, so there's no use in restricting this.
```suggestion
    pub fn new(node: HugrNodeID, port: hugr::Port) -> Self {
```